src/python/pants/option/options.py

```diff
@@ -389,35 +389,35 @@ def for_scope(
         :raises pants.option.errors.ConfigValidationError: if the scope is unknown.
         """
 
-        values_builder = OptionValueContainerBuilder()
-        flags_in_scope = self._scope_to_flags.get(scope, [])
-        parse_args_request = self._make_parse_args_request(flags_in_scope, values_builder)
-        legacy_values = self.get_parser(scope).parse_args(
-            parse_args_request, log_warnings=log_parser_warnings
-        )
+        native_values = self.get_parser(scope).parse_args_native(self._native_parser)
         native_mismatch_msgs = []
 
         if self._native_options_validation == NativeOptionsValidation.ignore:
-            native_values = None
+            legacy_values = None
         else:
             try:
-                native_values = self.get_parser(scope).parse_args_native(self._native_parser)
+                values_builder = OptionValueContainerBuilder()
+                flags_in_scope = self._scope_to_flags.get(scope, [])
+                parse_args_request = self._make_parse_args_request(flags_in_scope, values_builder)
+                legacy_values = self.get_parser(scope).parse_args(
+                    parse_args_request, log_warnings=log_parser_warnings
+                )
             except Exception as e:
                 native_mismatch_msgs.append(
                     f"Failed to parse options with native parser due to error:\n    {e}"
```

Comment:
```suggestion
                    f"Failed to parse options with legacy parser due to error:\n    {e}"
```

Reply:
Not a nitpick, that was actually wrong. Fixed, but also looking into the test failures, which are due to things either not failing when they should (presumably because the checks happen later or not at all on the native parser) or just failing with a slightly different error message.